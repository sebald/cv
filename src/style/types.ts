import { Interpolation, ThemeProps } from 'styled-components';
import { theme } from './theme';

export type Theme = typeof theme;

export type ThemedInterpolation = Interpolation<ThemeProps<Theme>>;
