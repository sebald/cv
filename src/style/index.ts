export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
