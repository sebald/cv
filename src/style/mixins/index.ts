export * from './animation';
export * from './breakpoints';
export * from './color';
export * from './space';
export * from './typography';
export * from './width';

export * from './types';
