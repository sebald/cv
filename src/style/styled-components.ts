import * as styledComponents from 'styled-components';
import {
  ThemedStyledComponentsModule,
  ThemedStyledFunction,
} from 'styled-components';

import { Theme } from './types';

// https://github.com/styled-components/styled-components/issues/630#issuecomment-317172454
function withProps<U>() {
  return <P, T, O>(
    fn: ThemedStyledFunction<P, T, O>
  ): ThemedStyledFunction<P & U, T, O & U> =>
    fn as ThemedStyledFunction<P & U, T, O & U>;
}

// Alternative, maybe better approach?
type TTag = keyof JSX.IntrinsicElements;

function suit<U>(tag: TTag) {
  return styled[tag] as ThemedStyledFunction<
    JSX.IntrinsicElements[TTag] & U,
    Theme
  >;
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export { styled, css, injectGlobal, keyframes, withProps, suit, ThemeProvider };
