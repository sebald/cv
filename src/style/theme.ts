export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
