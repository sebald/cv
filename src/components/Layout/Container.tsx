import { media, styled } from '../../style';

export const Container = styled.div`
  display: flex;
  flex-direction: column;

  height: 100vh;
  width: 100%;

  ${media('small')`
    flex-wrap: wrap;
  `};

  @media print {
    flex-direction: row;
    flex-wrap: wrap;
  }
`;
