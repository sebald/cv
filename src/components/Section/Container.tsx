import * as React from 'react';

import { px, styled } from '../../style';
import { SectionHeading } from './SectionHeading';
import { SectionContainerProps } from './types';

const ContainerWrapper = styled.section`
  max-width: ${(props) => px(props.theme.layout.content)};
`;

export const Container: React.FC<SectionContainerProps> = ({
  className,
  title,
  children,
}) => (
  <ContainerWrapper className={className}>
    <SectionHeading>{title}</SectionHeading>
    {children}
  </ContainerWrapper>
);

Container.displayName = 'SectionContainer';
