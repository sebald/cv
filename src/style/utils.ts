import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
