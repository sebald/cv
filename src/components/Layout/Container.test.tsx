import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { Container } from './Container';

test('render correctly', () => {
  const wrapper = shallow(<Container theme={theme}>Text</Container>);
  expect(wrapper).toMatchSnapshot();
});
