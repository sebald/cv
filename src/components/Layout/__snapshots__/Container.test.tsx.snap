// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
}

@media all and (min-width:768px) {
  .c0 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media print {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

<div
  className="c0"
>
  Text
</div>
`;
